shuup/front/template_helpers/general.py

```diff
@@ -26,8 +26,28 @@
 from shuup.utils.translation import cache_translations_for_tree
 
 
-@contextfunction
-def get_listed_products(context, n_products, ordering=None, filter_dict=None, orderable_only=True):
+def _group_list_items(group_list, number):
+    for i in range(0, len(group_list), number):
+        yield tuple(group_list[i: i + number])
+
+
+def _is_sale_item(product, pricing_context):
+    if product.is_variation_parent():
+        for variation in product.variation_children.all():
+            if variation.get_price_info(pricing_context).is_discounted:
+                return True
+
+    elif product.is_package_parent():
+        for package_child in product.get_all_package_children():
+            if package_child.get_price_info(pricing_context).is_discounted:
+                return True
+
+    return product.get_price_info(pricing_context).is_discounted
```

Comment:
would it be possible to use `shuup.campaigns.utils.matcher.get_matching_for_product` here instead of all this?

Reply:
No. Otherwise the discounts module won't be included. This will also add a direct dependence of Shuup Campaigns app here.